### DISCUSSION TITLE
Ability to use system installed versions

### DISCUSSION THREAD

**Discussion by stephanedaviet** [jdx/mise#1998] [Q&A] [answered]
I have a bunch of Java versions installed on my Linux. I would like to be able to use them instead of installing a new one in the mise install folder. Is it possible? Does making a symlink from system installation folder to `.local/share/mise/installs` is enough and the right way to do it?

I know for the `system` keyword in `.tool-versions`, but it's only able to use the _default_ system version, not a specific one.

Here the versions I have:
```
$ update-alternatives --list java
/usr/lib/jvm/java-11-openjdk-amd64/bin/java
/usr/lib/jvm/java-17-openjdk-amd64/bin/java
/usr/lib/jvm/java-21-openjdk-amd64/bin/java
/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
/usr/lib/jvm/temurin-11-jdk-amd64/bin/java
/usr/lib/jvm/temurin-17-jdk-amd64/bin/java
/usr/lib/jvm/temurin-21-jdk-amd64/bin/java
/usr/lib/jvm/temurin-8-jdk-amd64/bin/java
```

**Comment by jdx** [comment]
use the path scope: https://mise.jdx.dev/configuration.html#scopes

**Comment by stephanedaviet** [comment]
It's not really portable across developers, as each person can have a different installation path. I solve the problem with symbolic links:
```
$ ls -l ~/.local/share/mise/installs/java/
[...] 11 -> /usr/lib/jvm/temurin-11-jdk-amd64/
[...] 17 -> /usr/lib/jvm/temurin-17-jdk-amd64/
[...] 21 -> /usr/lib/jvm/temurin-21-jdk-amd64/
[...] 8 -> /usr/lib/jvm/temurin-8-jdk-amd64/
```

This way, a `java 11` in `.tools-version` use the system installed version instead of downloading a new one.

**Reply by jdx** [reply, marked as answer]
You can use aliases or .mise.local.toml

**Reply by jdx** [reply]
Symlinks are fine too of course
